Detrend Speedup

kmharrington opened the pull request:
The hour+ SAT books are giving us a good opportunity to profile the `tod_ops` code. There are several we'll need to work on but this is the start. 

Detrend on 1 UFM x 1.2 hours of data was taking ~110 seconds on simons1. The longest parts were subtracting the slopes and recasting at then end (in case data types had changed). Moving to the `-=`, using the `for` loop if possible, and only recasting if the data type had changed got it down to ~6 seconds. Added some comments so we hopefully remember why it's done this way.

Also add another argument to `get_glitch_flags` which lets us speed that up a little bit but that one will take some more careful work to speed up in the future (it's currently at ~240 seconds). 

Edit: Added some quick changes to `calc_psd` to let us just reduce the samples we call for it. I'm not planning any other changes on this branch.

mhasself reviewed (changes requested):
Looks fast but the docstring needs to be updated -- this now does modify the signal in place.  Does that break any in-tree usage?  I'm ok with breaking the default behavior, but it needs to be propagated.

Related augmentation proposal: add a "wrap=None" argument, which by default will simply detrend the signal in place ... but if passed as a string (wrap="signal_detrended") will cause the input data to be copied and stored in the axisman under that key.

mhasself commented on sotodlib/tod_ops/detrend.py:
```diff
@@ -52,14 +53,24 @@ def detrend_data(tod, method='linear', axis_name='samps',
         signal = signal.transpose(tuple(axis_reorder))
-    return signal.astype(dtype_in)
+
+    if signal.dtype != dtype_in:
+        # .astype takes awhile on long arrays, only cast if necessary
+        return signal.astype(dtype_in)
+    else:
+        return signal
```
It might be worth guarding against the first case entirely.  If you construct x and slopes to have the same type as signal, that should be enough.  Then you can assert signal.dtype == dtype_in as a fail-safe that will hopefully never ring.

kmharrington commented:
Ok @mhasself  I think I've addressed your comments. In the process I've realized that we've gotten to the point where `detrend_data` and `detrend_tod` had effectively become redundant, so I've renamed `detrend_data` to `detrend_tod`. (most people outside of sotodlib have been using `detrend_tod`, so I expect that's the least disruptive. 

I've updated the tests as well.

mhasself commented on sotodlib/tod_ops/detrend.py:
```diff
@@ -1,10 +1,19 @@
-    """Returns detrended data. Decide for yourself if it goes
-        into the axis manager. Generally intended for use before filter.
+    """Returns detrended data. Detrends data in place by default but pass
+        in_place=False if you would like a copied array (such as if you're just
```
Un-indent second (and all subsequent) lines of docstring.

In argument descriptions, please capitalize start of sentences after periods.

The docstring style here isn't quite numpy nor google... but sphinx seems ok with it, and it matches the style of other tod_ops docs.  So I won't fight that.

mhasself commented on sotodlib/tod_ops/detrend.py:
```diff
@@ -49,28 +66,26 @@ def detrend_data(tod, method='linear', axis_name='samps',
         count = max(1, min(count, signal.shape[-1] // 2))
-        slopes = signal[...,-count:].mean(axis=-1)-signal[...,:count].mean(axis=-1)
-        signal = signal - slopes[...,None] * x
+        slopes = signal[...,-count:].mean(axis=-1,dtype=dtype_in)-signal[...,:count].mean(axis=-1,dtype=dtype_in)
```
Please flake8 this line... the suppression of spaces has not, alas, led to it all being readable on a single line.

Other lines below have arithmetic spacing violations too.

mhasself reviewed (approved):
Happy to see those detrend functions merged!  I'd appreciate some attention to formatting though ... see comments on a couple lines, but there is trouble of this sort throughout tod_ops.  If we'd rather do a bulk reformat in separate PR, I will tolerate a punt on this.
